code/src/app/api/carInfo/route.ts

```diff
@@ -0,0 +1,364 @@
+ * @param {string} req.body.year - The year of the car.
+ * @param {string} req.body.license - The license number of the car.
+ * @param {string} req.body.mileage - The current mileage of the car.
+ * @param {string} req.body.model - The model of the car.
+ * @param {string} req.body.color - The color of the car.
+ * @param {string} req.body.seats - The number of seats in the car.
+ * @param {string} req.body.condition - The condition of the car.
+ * @param {string} req.body.mileageLastOilChange - The mileage of last oil change.
+ * @param {string} req.body.mileageLastTireChange - The mileage of last tire change.
+ * @param {string} req.body.dateNextOilChange - The date of next oil change.
+ * @param {string} req.body.dateNextTireChange - The date of next tire change.
+ * @returns {Promise<NextResponse>} A response object.
+ * 201: Car added successfully. New car info returned.
+ * 500: Unexpected server error.
+ * @throws {Error} Throws an error if there's an issue with the registration process.
+ */
+
+export async function POST(req: Request) {
+  try {
+    const {
+      manufacturer,
+      type,
+      year,
+      license,
+      mileage,
+      model,
+      color,
+      seats,
+      condition,
+      mileageLastOilChange,
+      mileageLastTireChange,
+      dateNextOilChange,
+      dateNextTireChange,
+    } = await req.json();
+
+    // Set defaults
+    const oilOverdue = false;
+    const tireOverdue = false;
+    const maintenanceRequired = false;
+
+    await dbConnect();
+
+    const existingLicense = await car.findOne({ license });
+    if (existingLicense) {
+      return NextResponse.json(
+        { message: 'License number already registered.' },
+        { status: 400 },
+      );
+    }
+
+    const newCar = await car.create({
+      manufacturer,
+      type,
+      year,
+      license,
+      mileage,
+      model,
+      color,
+      seats,
+      condition,
+      mileageLastOilChange,
+      mileageLastTireChange,
+      dateNextOilChange,
+      dateNextTireChange,
+      oilOverdue,
+      tireOverdue,
+      maintenanceRequired,
+    });
+
+    return NextResponse.json({ newCar }, { status: 201 });
+  } catch (err: any) {
+    return NextResponse.json(
+      { message: `Server Error: ${err}` },
+      { status: 500 },
+    );
+  }
+}
+
+/**
+ * Get car info REST API access.
+ * This endpoint allows a user to get car information.
+ *
+ * @async
+ * @function
+ * @param {Object} req - The request object.
+ * @param {Object} req.url - The url of the request.
+ * @returns {Promise<NextResponse>} A response object.
+ * 200: Car retrieved successfully.
+ * 400: Car ID not provided or invalid.
+ * 404: Car ID not found in DB.
+ * 500: Unexpected server error.
+ * @throws {Error} Throws an error if there's an issue with the registration process.
+ */
+
+export async function GET(req: Request) {
+  try {
+    const url = new URL(req.url);
+    const carId = url.searchParams.get('carId');
```

Comment:
carId is the mongoDB autogenerated _id. Can be updated later, this is for quick setup